operator/controllers/agent/reconciler.go

```diff
@@ -447,24 +439,24 @@ func (r *AgentReconciler) checkDefaults(ctx context.Context, instance *agentv1al
 		instance.Spec.ConfigSpec.FluxNinja.InstallationMode = "KUBERNETES_DAEMONSET"
 	}
 
-	if instance.Spec.Secrets.FluxNinjaExtension.Create && instance.Spec.Secrets.FluxNinjaExtension.Value == "" {
-		instance.Status.Resources = controllers.FailedStatus
-		r.Recorder.Eventf(instance, corev1.EventTypeWarning, "ValidationFailed", "The value for 'spec.secrets.fluxNinjaExtension.value' can not be empty when 'spec.secrets.fluxNinjaExtension.create' is set to true")
-		errUpdate := r.updateStatus(ctx, instance)
-		if errUpdate != nil {
-			return errUpdate
+	if !instance.Spec.ConfigSpec.FluxNinja.EnableCloudController {
+		if len(instance.Spec.ConfigSpec.Etcd.Endpoints) == 0 {
+			return updateStatus(instance, "ValidationFailed", "At least one etcd endpoint must be provided under spec.config.etcd.endpoints.")
 		}
-		return nil
+
+		if instance.Spec.ConfigSpec.Prometheus.Address == "" {
+			return updateStatus(instance, "ValidationFailed", "The address for Prometheus must be provided under spec.config.prometheus.address.")
+		}
+	} else if instance.Spec.ConfigSpec.FluxNinja.Endpoint == "" {
+		return updateStatus(instance, "ValidationFailed", "The endpoint for Flux Ninja must be provided under spec.config.fluxNinja.endpoint.")
+	}
```

Comment:
Nit: can we somehow reuse the code from hook.go?

Reply:
The checks are same but the return and all are different there so for now let's keep them as is and we can pick this later